src/analyzing/Visit_pattern_ids.ml

```diff
@@ -0,0 +1,51 @@
+(* Brandon Wu
+ *
+ * Copyright (C) 2023 r2c
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file license.txt.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * license.txt for more details.
+ *)
+
+module G = AST_generic
+
+class ['self] pat_id_visitor =
+  object (_self : 'self)
+    inherit ['self] AST_generic.iter_no_id_info as super
+
+    method! visit_pattern store pat =
+      match pat with
+      | PatAs (_, (id, id_info))
+      | PatId (id, id_info) ->
+          super#visit_pattern store pat;
+          Common.push (id, id_info) store
+      | PatLiteral _
+      | PatConstructor _
+      | PatRecord _
+      | PatTuple _
+      | PatList _
+      | PatKeyVal _
+      | PatUnderscore _
+      | PatDisj _
+      | PatTyped _
+      | PatWhen _
+      | PatType _
+      | PatEllipsis _
+      | DisjPat _
+      | OtherPat _ ->
```

Comment:
In Deep_tainting we had to handle this `OtherPat` case specially https://github.com/returntocorp/semgrep-proprietary/pull/833/files#diff-c1d42c3d0b66c474284f82bb2b7fad3ab670965fee3a6cade993129f63320877L324 (not sure why it wasn't a `PatRecord` 🤷‍♂️), and then traverse the fields of the record https://github.com/returntocorp/semgrep-proprietary/pull/833/files#diff-c1d42c3d0b66c474284f82bb2b7fad3ab670965fee3a6cade993129f63320877L332. Have you tried this in Pro?

Reply:
Oh, I thought that was subsumed under the `PatRecord` case, but I only just noticed it's an expression record and not a PatRecord. I just restored that case here, in the most recent commit.